lib/overcommit/hook/pre_commit/ruby_syntax.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Overcommit::Hook::PreCommit
+  # Runs `ruby -c` against all Ruby files.
+  #
+  class RubySyntax < Base
+    MESSAGE_TYPE_CATEGORIZER = lambda do |type|
+      type.match?(/^(syntax)?\s*error/) ? :error : :warning
+    end
+
+    def run
+      cmd = ['bash', '-c', 'find . -name "*.rb" -exec ruby -c {} \; | grep -q "Syntax OK"']
```

Comment:
Note that the way you've structured this means that the command will run against every `*.rb` file in your repository. `pre-commit` hooks should run against only the files that were modified. 

You should be using the `include` option of the `RubySyntax` hook configuration to specify this (which you do), so it's not clear to me why you are specifying the command you have here.

Reply:
Yes, you are right, my last commit fixes that. Only changed files will be checked.